test/cbmc/proofs/dataHandlerCleanup/dataHandlerCleanup_harness.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * AWS IoT Over-the-air Update v3.1.0
+ * Copyright (C) 2021 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file dataHandlerCleanup_harness.c
+ * @brief Implements the proof harness for dataHandlerCleanup function.
+ */
+/* Include headers for ota agent. */
+#include "ota.h"
+#include "stubs.h"
+
+extern OtaAgentContext_t otaAgent;
+extern void dataHandlerCleanup( void );
+
+void dataHandlerCleanup_harness()
+{
+    OtaInterfaces_t otaInterface;
+
+    otaInterface.os.timer.stop = stopTimerStub;
+    otaInterface.os.event.send = sendEventStub;
+
+    /* otaAgent.pOtaInterface can never be NULL as it is always checked at the start of the OTA
+     * Agent specifically in receiveAndProcessOTAEvent function.*/
+    otaAgent.pOtaInterface = &otaInterface;
+
+    ( void ) dataHandlerCleanup();
```

Comment:
Is there anything we can check to see if it works as exepected?

Reply:
The only way to check if this works as expected is to check the coverage report of the proof. It would indicate which functions are skipped and which lines are hit/missed.

Reply:
Why is the output (return value) being casted to `void` even when the return type is already `void`?